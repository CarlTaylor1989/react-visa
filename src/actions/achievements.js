import * as types from './achievementTypes';

export function setCompletedAchievement(achievement) {
  return {
    type: types.SET_COMPLETED_ACHIEVEMENT,
    achievement
  };
}

export function restoreAchievements(completed, lastCompleted) {
  return {
    type: types.RESTORE_ACHIEVEMENTS,
    completed,
    lastCompleted
  };
}

export function updateProductAchievements() {
  return {
    type: types.UPDATE_PRODUCT_ACHIEVEMENTS
  };
}

export function updateProposalAchievements() {
  return {
    type: types.UPDATE_PROPOSAL_ACHIEVEMENTS
  };
}

export function updateScoredAchievements() {
  return {
    type: types.UPDATE_SCORED_ACHIEVEMENTS
  };
}

export function updateSettingsAchievements() {
  return {
    type: types.UPDATE_SETTINGS_ACHIEVEMENTS
  };
}

export function updateTimedAchievements() {
  return {
    type: types.UPDATE_TIMED_ACHIEVEMENTS
  };
}

export function resetChangedAchievement() {
  return {
    type: types.RESET_CHANGED_ACHIEVEMENT
  };
}

export function resetAchievements() {
  return {
    type: types.RESET_ACHIEVEMENTS
  };
}
