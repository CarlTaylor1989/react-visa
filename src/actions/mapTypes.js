export const SET_CHALLENGES = 'SET_CHALLENGES';
export const SET_FAMILIES = 'SET_FAMILIES';
export const MAP_STATE_INITIALISED = 'MAP_STATE_INITIALISED';

export const UPDATE_CHALLENGE_STATUS = 'UPDATE_CHALLENGE_STATUS';
export const UPDATE_FAMILY_STATUS = 'UPDATE_FAMILY_STATUS';

export const SET_LAST_CHALLENGES_MODIFIED = 'SET_LAST_CHALLENGES_MODIFIED';
export const SET_LAST_PRODUCT_COMPLETED = 'SET_LAST_PRODUCT_COMPLETED';
export const RESET_LAST_PRODUCT_COMPLETED = 'RESET_LAST_PRODUCT_COMPLETED';

export const SET_CURRENT_FAMILY_PRODUCT = 'SET_CURRENT_FAMILY_PRODUCT';

export const ADD_PERFECT_CHALLENGE = 'ADD_PERFECT_CHALLENGE';
export const SET_COMPLETED_PRODUCTS = 'SET_COMPLETED_PRODUCTS';
export const ADD_COMPLETED_PRODUCT = 'ADD_COMPLETED_PRODUCT';
export const SET_COMPLETED_CHALLENGES = 'SET_COMPLETED_CHALLENGES';
export const ADD_COMPLETED_CHALLENGE = 'ADD_COMPLETED_CHALLENGE';

export const RESET_PROGRESS = 'RESET_PROGRESS';

export const SET_PRODUCT_PAGE_VIEWED_SESSION = 'SET_PRODUCT_PAGE_VIEWED_SESSION';
