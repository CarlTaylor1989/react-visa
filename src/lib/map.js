import _ from 'lodash';
import {
  CHALLENGE_UNAVAILABLE_STATUS,
  CHALLENGE_IN_PROGRESS_STATUS,
  CHALLENGE_AVAILABLE_STATUS,
  CHALLENGE_FAILED_STATUS,
  CHALLENGE_COMPLETION_STATUS,
  CHALLENGE_PERFECT_STATUS,
  CHALLENGE_NOT_STARTED_CLASS,
  CHALLENGE_AVAILABLE_CLASS,
  CHALLENGE_COMPLETED_CLASS,
  CHALLENGE_PERFECT_CLASS,
  FAMILY_INCOMPLETE_STATUS,
  FAMILY_COMPLETION_STATUS,
  SCO_COMPLETION_PASSED,
  SCO_COMPLETION_FAILED,
  SCO_COMPLETION_UNKNOWN,
  SCO_PERFECT_SCORE,
  XAPI_FAIL_CHALLENGE,
  XAPI_FAIL_CHALLENGE_SCORE,
  XAPI_COMPLETED_CHALLENGE,
  XAPI_COMPLETED_CHALLENGE_SCORE,
  XAPI_PERFECT_CHALLENGE,
  XAPI_PERFECT_CHALLENGE_SCORE
} from '../config/constants';

export const lib = {};

lib.determineFamilyStatus = (data) => {
  let status;
  if (data.failed || data.inprogress || data.available || data.unavailable) {
    status = FAMILY_INCOMPLETE_STATUS;
  } else {
    status = FAMILY_COMPLETION_STATUS;
  }
  return status;
};

lib.getStoredChallengeStatus = (data) => {
  const challenges = {
    [data.id]: {}
  };

  if (data.perfect) {
    data.perfect.forEach((ch) => {
      challenges[data.id][ch] = CHALLENGE_PERFECT_STATUS;
    });
  }

  if (data.completed) {
    data.completed.forEach((ch) => {
      challenges[data.id][ch] = CHALLENGE_COMPLETION_STATUS;
    });
  }

  if (data.failed) {
    data.failed.forEach((ch) => {
      challenges[data.id][ch] = CHALLENGE_FAILED_STATUS;
    });
  }

  if (data.inprogress) {
    data.inprogress.forEach((ch) => {
      challenges[data.id][ch] = CHALLENGE_IN_PROGRESS_STATUS;
    });
  }

  if (data.available) {
    data.available.forEach((ch) => {
      challenges[data.id][ch] = CHALLENGE_AVAILABLE_STATUS;
    });
  }

  if (data.unavailable) {
    data.unavailable.forEach((ch) => {
      challenges[data.id][ch] = CHALLENGE_UNAVAILABLE_STATUS;
    });
  }

  return challenges;
};

export const getStoredMapData = (suspendData) => {
  const families = [];
  let challenges = {};

  if (suspendData && suspendData.map && suspendData.map.families
    && suspendData.map.families.length) {
    suspendData.map.families.forEach((family) => {
      families.push({
        id: family.id,
        status: lib.determineFamilyStatus(family)
      });

      challenges = {
        ...challenges,
        ...lib.getStoredChallengeStatus(family)
      };
    });
  }

  return {
    families,
    challenges
  };
};

/**
 * Gets the challenge status from the stored data
 * @param {array} challenges A list of stored challenges
 * @param {string} challengeId Challenge ID
 * @param {string} familyId Family ID
 * @param {boolean} isFirst Is the challenge the first in the product line
 * @return {number}
 */
export const getStoredChallengeStatus = (challenges, challengeId, familyId, isFirst) => {
  let status;

  if (challenges[familyId] && challenges[familyId][challengeId] !== undefined) {
    status = challenges[familyId][challengeId];
  } else if (isFirst) {
    status = CHALLENGE_AVAILABLE_STATUS;
  } else {
    status = CHALLENGE_UNAVAILABLE_STATUS;
  }

  return status;
};

/**
 * Determines the challenge status from the SCORM data
 * @param {string} status The completion status
 * @param {number} score The SCO score
 * @return {number}
 */
export const determineChallengeStatus = (status, score) => {
  let challengeStatus = CHALLENGE_AVAILABLE_STATUS;
  if (status === SCO_COMPLETION_PASSED) {
    if (score === SCO_PERFECT_SCORE) {
      challengeStatus = CHALLENGE_PERFECT_STATUS;
    } else {
      challengeStatus = CHALLENGE_COMPLETION_STATUS;
    }
  } else if (status === SCO_COMPLETION_FAILED) {
    challengeStatus = CHALLENGE_FAILED_STATUS;
  } else if (status === SCO_COMPLETION_UNKNOWN) {
    challengeStatus = CHALLENGE_IN_PROGRESS_STATUS;
  }
  return challengeStatus;
};

/**
 * Determines the class of an element and whether it's disabled or not based on the challeng status
 * @param {number} status The current challenge status
 * @return {object} An object with the class name and the disabled status
 */
export const getChallengeStatus = (status) => {
  let className = '';
  let disabled = false;
  switch (status) {
    case CHALLENGE_UNAVAILABLE_STATUS:
      className = CHALLENGE_NOT_STARTED_CLASS;
      disabled = true;
      break;
    case CHALLENGE_AVAILABLE_STATUS:
    case CHALLENGE_IN_PROGRESS_STATUS:
    case CHALLENGE_FAILED_STATUS:
      className = CHALLENGE_AVAILABLE_CLASS;
      break;
    case CHALLENGE_COMPLETION_STATUS:
      className = CHALLENGE_COMPLETED_CLASS;
      break;
    case CHALLENGE_PERFECT_STATUS:
      className = CHALLENGE_PERFECT_CLASS;
      break;
    default:
      className = CHALLENGE_NOT_STARTED_CLASS;
      break;
  }
  return {
    className,
    disabled
  };
};

/**
 * Returns the next available challenge index
 * @param {array} challenges A list of challenges
 * @return {number}
 */
export const getNextChallengeIndex = (challenges) => {
  const found = _.findIndex(challenges, ch => (
    ch.status === CHALLENGE_AVAILABLE_STATUS
    || ch.status === CHALLENGE_IN_PROGRESS_STATUS
    || ch.status === CHALLENGE_FAILED_STATUS
  ));
  return found < 0 ? challenges.length - 1 : found;
};

/**
 * Returns the challenge source data
 * @param {object} challenge The challenge from the game state
 * @param {object} challenges A dictionary of families with list of challenges
 */
export const findChallengeInChallenges = (challenge, challenges) => (
  challenges[challenge.familyId].find(ch => (
    ch.id === challenge.id && ch.productId === challenge.productId
  ))
);

/**
 * Gets the  status and score for the xAPI statement
 * @param {number} challengeStatus The challenge status
 * @return {object}
 */
export const getChallengeXAPIStatus = (status) => {
  const xApiStatus = {
    response: '',
    score: XAPI_FAIL_CHALLENGE_SCORE,
    success: true
  };

  switch (status) {
    case CHALLENGE_COMPLETION_STATUS:
      xApiStatus.response = XAPI_COMPLETED_CHALLENGE;
      xApiStatus.score = XAPI_COMPLETED_CHALLENGE_SCORE;
      break;
    case CHALLENGE_PERFECT_STATUS:
      xApiStatus.response = XAPI_PERFECT_CHALLENGE;
      xApiStatus.score = XAPI_PERFECT_CHALLENGE_SCORE;
      break;
    case CHALLENGE_UNAVAILABLE_STATUS:
    case CHALLENGE_AVAILABLE_STATUS:
    case CHALLENGE_IN_PROGRESS_STATUS:
    case CHALLENGE_FAILED_STATUS:
    default:
      xApiStatus.response = XAPI_FAIL_CHALLENGE;
      xApiStatus.score = XAPI_FAIL_CHALLENGE_SCORE;
      xApiStatus.success = false;
      break;
  }

  return xApiStatus;
};
