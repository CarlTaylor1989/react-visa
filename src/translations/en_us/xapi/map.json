{
  "title": "The Product Network Map",
  "description": "The main area learners engage with in the game. Here the learner’s goal is to work their way from the centre of the map to the end of each Line, which represent Visa product families (e.g. Identity and Security). Each family includes a number of product rails learners must complete by taking a sequence of mini-challenges that test the learner’s understanding of that product. Each product family and zone has an associated color, just like a metro map. The learner can start their journey from any point, taking a series of sequential challenges to ‘complete’ products and score points."
}
