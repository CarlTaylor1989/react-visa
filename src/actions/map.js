import * as types from './mapTypes';

export function setChallenges(challenges) {
  return {
    type: types.SET_CHALLENGES,
    challenges
  };
}

export function setFamilies(families) {
  return {
    type: types.SET_FAMILIES,
    families
  };
}

export function initialiseMapState(initialised) {
  return {
    type: types.MAP_STATE_INITIALISED,
    initialised
  };
}

export function updateChallengeStatus(id, familyId, status) {
  return {
    type: types.UPDATE_CHALLENGE_STATUS,
    id,
    familyId,
    status
  };
}

export function updateFamilyStatus(id, status) {
  return {
    type: types.UPDATE_FAMILY_STATUS,
    id,
    status
  };
}

export function setLastChallengesModified(lastChallengesModified) {
  return {
    type: types.SET_LAST_CHALLENGES_MODIFIED,
    lastChallengesModified
  };
}

export function setLastProductCompleted(lastProductCompleted) {
  return {
    type: types.SET_LAST_PRODUCT_COMPLETED,
    lastProductCompleted
  };
}

export function resetLastProductCompleted() {
  return {
    type: types.RESET_LAST_PRODUCT_COMPLETED
  };
}

export function setCurrentFamilyProduct(currentMap, currentFamily, currentProduct) {
  return {
    type: types.SET_CURRENT_FAMILY_PRODUCT,
    currentMap,
    currentFamily,
    currentProduct
  };
}

export function addPerfectChallenge(challenge) {
  return {
    type: types.ADD_PERFECT_CHALLENGE,
    challenge
  };
}

export function setCompletedProducts(products) {
  return {
    type: types.SET_COMPLETED_PRODUCTS,
    products
  };
}

export function addCompletedProduct(productId) {
  return {
    type: types.ADD_COMPLETED_PRODUCT,
    productId
  };
}

export function setCompletedChallenges(challenges) {
  return {
    type: types.SET_COMPLETED_CHALLENGES,
    challenges
  };
}

export function addCompletedChallenge(challengeId) {
  return {
    type: types.ADD_COMPLETED_CHALLENGE,
    challengeId
  };
}

export function resetProgress() {
  return {
    type: types.RESET_PROGRESS
  };
}

export function setProductPageViewedSession() {
  return {
    type: types.SET_PRODUCT_PAGE_VIEWED_SESSION
  };
}
