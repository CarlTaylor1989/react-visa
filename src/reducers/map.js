import { createReducer } from 'redux-create-reducer';
import { cloneDeep } from 'lodash';
import * as types from '../actions/mapTypes';

export const initialState = {
  challenges: [],
  families: [],
  initialised: false,
  lastChallengesModified: [],
  lastProductCompleted: '',
  lastProductModified: '',
  perfect: [],
  currentMap: '',
  currentFamily: '',
  currentProduct: '',
  completedProducts: [],
  completedChallenges: [],
  completedChallengesSession: false,
  completedProductsSession: false,
  productPageViewedSession: false
};

export const mapData = createReducer(cloneDeep(initialState), {
  [types.SET_CHALLENGES](state, action) {
    return {
      ...state,
      challenges: [
        ...state.challenges,
        ...action.challenges
      ]
    };
  },
  [types.SET_FAMILIES](state, action) {
    return {
      ...state,
      families: [
        ...state.families,
        ...action.families
      ]
    };
  },
  [types.MAP_STATE_INITIALISED](state, action) {
    return {
      ...state,
      initialised: action.initialised
    };
  },
  [types.UPDATE_CHALLENGE_STATUS](state, action) {
    return {
      ...state,
      challenges: state.challenges.map(challenge => (
        challenge.id === action.id
        && challenge.familyId === action.familyId ? {
            ...challenge,
            status: challenge.status < action.status ? action.status : challenge.status
          } : challenge))
    };
  },
  [types.UPDATE_FAMILY_STATUS](state, action) {
    return {
      ...state,
      families: state.families.map(family => (
        family.id === action.id ? {
          ...family,
          status: action.status
        } : family))
    };
  },
  [types.SET_LAST_CHALLENGES_MODIFIED](state, action) {
    return {
      ...state,
      lastChallengesModified: action.lastChallengesModified
    };
  },
  [types.SET_LAST_PRODUCT_COMPLETED](state, action) {
    return {
      ...state,
      lastProductCompleted: action.lastProductCompleted
    };
  },
  [types.RESET_LAST_PRODUCT_COMPLETED](state) {
    return {
      ...state,
      lastProductCompleted: initialState.lastProductCompleted
    };
  },
  [types.SET_CURRENT_FAMILY_PRODUCT](state, action) {
    return {
      ...state,
      currentMap: action.currentMap,
      currentFamily: action.currentFamily,
      currentProduct: action.currentProduct
    };
  },
  [types.ADD_PERFECT_CHALLENGE](state, action) {
    if (!state.perfect.includes(action.challenge)) {
      return {
        ...state,
        perfect: [
          ...state.perfect,
          action.challenge
        ]
      };
    }
    return {
      ...state
    };
  },
  [types.SET_COMPLETED_PRODUCTS](state, action) {
    return {
      ...state,
      completedProducts: [
        ...state.completedProducts,
        ...action.products
      ]
    };
  },
  [types.ADD_COMPLETED_PRODUCT](state, action) {
    return {
      ...state,
      completedProducts: [
        ...state.completedProducts,
        action.productId
      ],
      completedProductsSession: true
    };
  },
  [types.SET_COMPLETED_CHALLENGES](state, action) {
    return {
      ...state,
      completedChallenges: [
        ...state.completedChallenges,
        ...action.challenges
      ]
    };
  },
  [types.ADD_COMPLETED_CHALLENGE](state, action) {
    return {
      ...state,
      completedChallenges: [
        ...state.completedChallenges,
        action.challengeId
      ],
      completedChallengesSession: true
    };
  },
  [types.RESET_PROGRESS]() {
    return cloneDeep(initialState);
  },
  [types.SET_PRODUCT_PAGE_VIEWED_SESSION](state) {
    return {
      ...state,
      productPageViewedSession: true
    };
  }
});

export default { mapData };
