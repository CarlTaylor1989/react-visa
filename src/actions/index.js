import { push } from 'connected-react-router';
import * as AchievementActions from './achievements';
import * as AudioActions from './audio';
import * as ClientActions from './client';
import * as DiagnosticActions from './diagnostic';
import * as GenericActions from './generic';
import * as HintsActions from './hints';
import * as MapActions from './map';
import * as PromptActions from './prompts';
import * as ScoreActions from './score';
import * as ScormActions from './scorm';
import * as SettingsActions from './settings';
import * as StreakActions from './streak';
import * as TutorialActions from './tutorial';

const ActionCreators = Object.assign(
  {},
  AchievementActions,
  AudioActions,
  ClientActions,
  DiagnosticActions,
  GenericActions,
  HintsActions,
  MapActions,
  PromptActions,
  ScoreActions,
  ScormActions,
  SettingsActions,
  StreakActions,
  TutorialActions,
  { push }
);

export default ActionCreators;
