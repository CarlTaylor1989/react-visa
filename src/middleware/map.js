import * as types from '../actions/mapTypes';
import * as mapActions from '../actions/map';
import * as promptActions from '../actions/prompts';
import {
  CHALLENGE_COMPLETION_STATUS,
  CHALLENGE_PERFECT_STATUS
} from '../config/constants';

export const setCompletedChallenges = ({ dispatch, getState }) => next => (action) => {
  if (action.type === types.UPDATE_CHALLENGE_STATUS
    && (action.status === CHALLENGE_COMPLETION_STATUS
      || action.status === CHALLENGE_PERFECT_STATUS)) {
    const mapData = getState().mapData;
    const challengeId = action.familyId + action.id;
    if (!mapData.completedChallenges.includes(challengeId)) {
      dispatch(mapActions.addCompletedChallenge(challengeId));
    }
  }

  if (action.type === types.SET_CHALLENGES) {
    const completedChallenges = action.challenges.filter(ch => (
      ch.status === CHALLENGE_COMPLETION_STATUS || ch.status === CHALLENGE_PERFECT_STATUS));
    const completedChallengeIds = completedChallenges.map(ch => ch.familyId + ch.id);
    dispatch(mapActions.setCompletedChallenges(completedChallengeIds));

    completedChallenges.filter(ch => ch.status === CHALLENGE_PERFECT_STATUS).forEach((ch) => {
      dispatch(mapActions.addPerfectChallenge(ch.familyId + ch.id));
    });
  }

  next(action);
};

export const setCompletedProducts = ({ dispatch, getState }) => next => (action) => {
  const updatedAction = action;

  if (action.type === types.SET_LAST_PRODUCT_COMPLETED) {
    const mapData = getState().mapData;
    const productId = mapData.currentFamily + action.lastProductCompleted;
    if (!mapData.completedProducts.includes(productId)) {
      dispatch(mapActions.addCompletedProduct(productId));
      dispatch(promptActions.addProductPrompt(action.lastProductCompleted));
    } else {
      updatedAction.lastProductCompleted = '';
    }
  }

  if (action.type === types.SET_CHALLENGES) {
    const completedProducts = action.challenges.filter(ch => (
      ch.isFinal
      && (ch.status === CHALLENGE_COMPLETION_STATUS || ch.status === CHALLENGE_PERFECT_STATUS)));
    const completedProductsIds = completedProducts.map(ch => ch.familyId + ch.productId);
    dispatch(mapActions.setCompletedProducts(completedProductsIds));
  }

  next(updatedAction);
};

export const perfectChallenge = ({ dispatch }) => next => (action) => {
  if (action.type === types.UPDATE_CHALLENGE_STATUS && action.status === CHALLENGE_PERFECT_STATUS) {
    dispatch(mapActions.addPerfectChallenge(action.familyId + action.id));
  }

  next(action);
};

export default [
  setCompletedChallenges,
  setCompletedProducts,
  perfectChallenge
];
