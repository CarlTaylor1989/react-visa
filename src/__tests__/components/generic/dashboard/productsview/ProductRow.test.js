/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import {
  ProductRow,
  mapStateToProps,
  mapDispatchToProps
} from '../../../../../components/generic/dashboard/productsview/ProductRow';
import * as actions from '../../../../../actions/map';
import * as types from '../../../../../actions/mapTypes';

let mockProps;
beforeEach(() => {
  mockProps = {
    closePopup: jest.fn(),
    completedProducts: [],
    familyId: 'm1f2',
    mapId: 'm1',
    productId: 'p3',
    productName: 'irrelevant',
    setCurrentFamilyProduct: jest.fn(),
    setScreenReferrer: jest.fn(),
    type: 'type2'
  };
});

test('renders the ProductRow component with the incomplete styles', () => {
  const wrapper = shallow(<ProductRow {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders the ProductRow component with the complete styles', () => {
  mockProps.completedProducts = ['m1f2p3'];
  const wrapper = shallow(<ProductRow {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('method onProductRowClick()', () => {
  // setup
  const wrapper = shallow(<ProductRow {...mockProps} />);

  // function under test
  wrapper.instance().onProductRowClick();

  // expectations
  expect(mockProps.setScreenReferrer).toHaveBeenCalledWith(expect.any(String));
  expect(mockProps.setCurrentFamilyProduct).toHaveBeenCalled();
  expect(mockProps.closePopup).toHaveBeenCalled();
});

test('should mapStateToProps', () => {
  const state = {
    mapData: {
      completedProducts: []
    }
  };
  const props = mapStateToProps(state);
  expect(props).toEqual({
    ...state.mapData
  });
});

test('should mapDispatchToProps', () => {
  // setup
  const currentMap = 'm1';
  const currentFamily = 'm1f1';
  const currentProduct = 'p1';
  const dispatch = jest.fn();
  jest.spyOn(actions, 'setCurrentFamilyProduct').mockReturnValue('setCurrentFamilyProduct()');
  const props = mapDispatchToProps(dispatch);

  // function under test
  props.setCurrentFamilyProduct(currentMap, currentFamily, currentProduct);

  // expectation
  expect(dispatch).toHaveBeenCalledWith({
    type: types.SET_CURRENT_FAMILY_PRODUCT,
    currentMap,
    currentFamily,
    currentProduct
  });
});
