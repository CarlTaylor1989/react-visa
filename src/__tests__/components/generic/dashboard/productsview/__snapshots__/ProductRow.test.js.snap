// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ProductRow component with the complete styles 1`] = `
<NavLink
  className="productRow type2 completed"
  onClick={[Function]}
  to="/game/network/product"
>
  <InlineSVG
    className="productRowIcon"
    element="i"
    raw={false}
    src=""
  />
  <div
    className="productName"
    dangerouslySetInnerHTML={
      Object {
        "__html": "irrelevant",
      }
    }
  />
  <Connect(ProductCompletion)
    familyId="m1f2"
    productId="p3"
  />
</NavLink>
`;

exports[`renders the ProductRow component with the incomplete styles 1`] = `
<NavLink
  className="productRow type2 "
  onClick={[Function]}
  to="/game/network/product"
>
  <InlineSVG
    className="productRowIcon"
    element="i"
    raw={false}
    src=""
  />
  <div
    className="productName"
    dangerouslySetInnerHTML={
      Object {
        "__html": "irrelevant",
      }
    }
  />
  <Connect(ProductCompletion)
    familyId="m1f2"
    productId="p3"
  />
</NavLink>
`;
